### ISSUE TITLE
change bucket name not working

### ISSUE THREAD

**Issue by pratik2pris** [TTLabs/EvaporateJS#292]
I am trying to upload multiple files with diff bucket name.

e.g.

$.each(this.filelist,function(){
      evaporate.add(file,{
          bucket : this.bucket_name
      })
});

But the plugin pick the last file bucket name. If I upload file one by one then it works fine.

More information : 

SignedS3AWSRequest.prototype.getPath = function () {
        alert(this.con.bucket);
    var path = '/' + this.con.bucket + '/' + this.fileUpload.name;
    if (this.con.cloudfront || this.awsUrl.indexOf('cloudfront') > -1) {
      path = '/' + this.fileUpload.name;
    }
    return path;

Above will return always the bucket name from the last file. I assume that this.con is global here when override. I think it should be something line this.file.config.bucket

Could you help me ?

**Comment by bikeath1337** [comment]
Thanks for the report! I found the issue and will submit a fix later today or tomorrow.

**Comment by pratik2pris** [comment]
Ok Thanks!

**Comment by mhenkel1** [comment]
You could of course also use a different put your ForEach function.
Right now you always reference "bucket" to "this.bucket_name", but usually the files are all added before the first upload will start. And since you are changing this.bucket_name somehow during adding the files, evaporate will use the reference to this.bucket_name when it starts the uploads.
However you can use a workaround, where you use a "promise-safe" ForEach function.

```
Array.prototype.forEach.call(files, function(file) {
  evaporate.add(file,{
    bucket : this.bucket_name
  })
});
```

I had the same problem, but I wasn't just using the bucket name, I was trying to add separate upload  class instances to each individual file upload, to show a progress bar and speeds for every upload. However evaporate would only use the latest added progress bar and the progress and speed would "jump around":

```

              Array.prototype.forEach.call(files, function(file) {
                var upload = {
                  name: file.name,
                  percent: 0,
                };
                t.uploads.push(upload);
                evaporate.add({
                  file: file,
                  name: t.uuid + '/' + file.name,
                  complete: function(r) {
                    upload.percent = 100;
                  }.bind(upload),
                  progress: function(progress, stats) {
                    upload.percent = progress * 100;
                  }.bind(upload),
                })
              });
```

**Comment by bikeath1337** [comment]
@mhenkel1 @pratik2pris I just published 2.0.6 which should address this issue. I added a spec to cover it as well. Thanks for the report.
